src/mongo/db/auth/oidc/oidc_commands.cpp

```diff
@@ -126,59 +130,73 @@ class OidcListKeys : public OidcKeysCommand<ActionType::oidcListKeys> {
 
 MONGO_REGISTER_COMMAND(OidcListKeys).forShard().forRouter();
 
-class OidcRefreshKeys : public OidcKeysCommand<ActionType::oidcRefreshKeys> {
+namespace {
+class JWKSRefreshingFailureContainer {
 public:
-    explicit OidcRefreshKeys() : OidcKeysCommand("oidcRefreshKeys") {}
+    bool empty() const noexcept {
+        return _failures.empty();
+    }
 
-private:
-    // Combines multiple Status objects into a single Status.
-    // The input vector must contain only non-OK statuses.
-    Status combineStatuses(const std::vector<Status>& statuses) {
-        if (statuses.empty()) {
-            return Status::OK();
-        }
+    void append(std::string issuer, Status status) {
+        _failures.emplace_back(std::move(issuer), std::move(status));
+    }
 
-        if (statuses.size() == 1) {
-            return statuses[0];
-        }
+    void serialize(BSONObjBuilder* bob) const;
 
-        StringBuilder combinedMessage;
-        combinedMessage << "Multiple errors occurred while refreshing:\n";
-        for (const auto& status : statuses) {
-            invariant(!status.isOK());
-            combinedMessage << status.toString() << "\n";
-        }
+private:
+    std::vector<std::pair<std::string, Status>> _failures;
+
+    static constexpr const char* kFailuresFieldName{"failures"};
```

Comment:
why not `static constexpr auto kFailuresFieldName = "failures"_sd;` ? I noticed this is commonly used approach especially when the string is only used as `StringData`.

Reply:
Fixed